subiquity/controllers/filesystem.py

```diff
@@ -130,25 +132,44 @@ def __init__(self, app):
         self.answers.setdefault('manual', [])
         self._cur_probe = None
         self._monitor = None
-        self.ui_shown = False
+        self._udev_listen_handle = None
 
     def start(self):
         self._start_probe(restricted=False)
         context = pyudev.Context()
         self._monitor = pyudev.Monitor.from_netlink(context)
         self._monitor.filter_by(subsystem='block')
         self._monitor.enable_receiving()
-        self._last_udev_event = time.time()
-        self.loop.watch_file(self._monitor.fileno(), self._udev_event)
+        self.start_listening_udev()
+
+    def start_listening_udev(self):
+        self._udev_listen_handle = self.loop.watch_file(
+            self._monitor.fileno(), self._udev_event)
+
+    def stop_listening_udev(self):
+        if self._udev_listen_handle is not None:
+            self.loop.remove_watch_file(self._udev_listen_handle)
+            self._udev_listen_handle = None
 
     def _udev_event(self):
-        action, dev = self._monitor.receive_device()
-        log.debug("_udev_event %s %s", action, dev)
-        if time.time() - self._last_udev_event > 0.1 and not self.ui_shown:
-            self._start_probe(restricted=False)
-        self._last_udev_event = time.time()
+        cp = run_command(['udevadm', 'settle', '-t', '0'])
+        if cp.returncode != 0:
```

Comment:
Hrm, if we have a non-empty queue (returncode != 0) shouldn't we call udevadm settle and then probe?

Reply:
That would block the UI, so no. We could do it in a background thread I guess but this seems to fail the effort / benefit ratio